tests/instruments/test_instrument.py

```diff
@@ -150,6 +151,41 @@ class Fake(FakeInstrument):
     assert fake.x == 5
 
 
+def test_control_preprocess_reply():
+    # test setting preprocess_reply at property-level
+    class Fake(FakeInstrument):
+        x = Instrument.control(
+            "", "JUNK%d", "",
+            preprocess_reply=lambda v: v.replace('JUNK', ''),
+            cast=int
+        )
+
+    fake = Fake()
+    fake.x = 5
+    assert fake.read() == 'JUNK5'
+    fake.x = 5
+    assert fake.x == 5
+    fake.x = 5
+    assert type(fake.x) == int
+
+    # test setting preprocess_reply at Adapter-level
```

Comment:
can you put this in a separate test, please?

Reply:
done